### PR TITLE
fix(telemetry): wrong requestId in UserDecision (Prod data polluted)

### PR DESCRIPTION
**Pull request by laileni-aws**
## Problem
Sending last recommendation request id for all the `userDecision` telemetry events in a session/trigger.
https://issues.amazon.com/issues/ConsolasIssue-12227
## Solution
Send `userDecision` events with respective requestId of each recommendation in a session/trigger.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]

File: src/codewhisperer/util/telemetryHelper.ts [hunk 1]
```diff
@@ -178,7 +178,7 @@ export class TelemetryHelper {
                 recommendationSuggestionState?.set(i, 'Empty')
             }
             const event: CodewhispererUserDecision = {
-                codewhispererRequestId: requestId,
+                codewhispererRequestId: requestIdList[i],
```
Comment (andrewyuq) [inline src/codewhisperer/util/telemetryHelper.ts:181]: Ideally it should be a list of recommenationContexts where in each context there's a recommendation, and a requestId(and other things related to that completion). So you don't have to maintain two separate list and worry about the potential problem of inconsistency between these two lists.

Comment (laileni-aws) [reply src/codewhisperer/util/telemetryHelper.ts:181]: What if there is no recommendation? but still we need to send the requestId? `recordUserDecisionTelemetryForEmptyList`. I think, in this case we can not map recommendation to requestId.
Correct me if I am wrong?

File: src/codewhisperer/service/recommendationHandler.ts [hunk 2]
```diff
@@ -83,6 +84,7 @@ export class RecommendationHandler {
 
     constructor() {
         this.requestId = ''
+        this.requestIdList = []
```
[the remark this answers is not in the document]
Comment (laileni-aws) [reply src/codewhisperer/service/recommendationHandler.ts]: Yeah, updated accordingly!

File: src/codewhisperer/util/telemetryHelper.ts (hunk 1 above)
Comment (andrewyuq) [reply src/codewhisperer/util/telemetryHelper.ts:181]: for that one you can just use the local requestId and pass that to the event, for the rest, we can add the recommendationContext to the session using similar logic in line 383 of the file recommendationHandler.ts

Comment (andrewyuq) [reply src/codewhisperer/util/telemetryHelper.ts:181]: This is not a blocker for now.

Comment (laileni-aws) [reply src/codewhisperer/util/telemetryHelper.ts:182]: Added a TODO for further changes.

**Review by andrewyuq — APPROVED** [review]

**Pull request merged**